lib/ts/types.ts

```diff
@@ -20,7 +20,7 @@ import { TypeFramework } from "./framework/types";
 
 export type AppInfo = {
     appName: string;
-    websiteDomain: string;
+    websiteDomain: string | ((userContext: any) => string | undefined);
```

Comment:
```suggestion
    websiteDomain: string | ((userContext: any) => Promise<string | undefined> | string | undefined);
```

Reply:
- This allows users to make any network call as well in the function if they want to.
- Also, if the user returns `undefined`, it will send backend a `400` error right?